libs/vkd3d/resource.c

```diff
@@ -3527,26 +3527,24 @@ bool vkd3d_create_texture_view(struct d3d12_device *device, const struct vkd3d_t
      * The clamp is absolute, and not affected by the baseMipLevel. */
     if (desc->miplevel_clamp <= (float)(desc->miplevel_idx + desc->miplevel_count - 1))
     {
-        if (device->device_info.image_view_min_lod_features.minLod)
+        if (desc->miplevel_clamp > (float)desc->miplevel_idx)
         {
-            min_lod_desc.sType = VK_STRUCTURE_TYPE_IMAGE_VIEW_MIN_LOD_CREATE_INFO_EXT;
-            min_lod_desc.pNext = NULL;
-            min_lod_desc.minLod = desc->miplevel_clamp;
-            vk_prepend_struct(&view_desc, &min_lod_desc);
-        }
-        else
-        {
-            if (desc->miplevel_clamp != 0.0f)
-                FIXME_ONCE("Cannot handle MinResourceLOD clamp of %f correctly.\n", desc->miplevel_clamp);
-            /* This is not correct, but it's the best we can do without VK_EXT_image_view_min_lod.
-            * It should at least avoid a scenario where implicit LOD fetches from invalid levels. */
-            if (desc->miplevel_clamp >= 1.0f)
+            if (device->device_info.image_view_min_lod_features.minLod)
             {
-                uint32_t clamp_base_level = max((uint32_t)desc->miplevel_clamp, view_desc.subresourceRange.baseMipLevel);
+                min_lod_desc.sType = VK_STRUCTURE_TYPE_IMAGE_VIEW_MIN_LOD_CREATE_INFO_EXT;
+                min_lod_desc.pNext = NULL;
+                min_lod_desc.minLod = desc->miplevel_clamp;
+                vk_prepend_struct(&view_desc, &min_lod_desc);
+            }
+            else
+            {
+                FIXME_ONCE("Cannot handle MinResourceLOD clamp of %f correctly.\n", desc->miplevel_clamp);
```

Comment:
This creates warnings because the variables below are suddenly in the same scope now.

Reply:
Fixed.